tests/model/DataObjectTest.php

```diff
@@ -16,18 +16,31 @@ class DataObjectTest extends SapphireTest {
 		'DataObjectTest_FieldlessSubTable',
 		'DataObjectTest_ValidatedObject',
 		'DataObjectTest_Player',
-		'DataObjectTest_TeamComment'
+		'DataObjectTest_TeamComment',
+		'DataObjectTest_ExtendedTeamComment'
 	);
 
-	public function testBaseFieldsExcludedFromDb() {
-		$obj = new DataObjectTest_ValidatedObject();
-
+	public function testDb() {
+		$obj = new DataObjectTest_TeamComment();
 		$dbFields = $obj->db();
+
+		// Assert fields are included
 		$this->assertArrayHasKey('Name', $dbFields);
+
+		// Assert the base fields are excluded
 		$this->assertArrayNotHasKey('Created', $dbFields);
 		$this->assertArrayNotHasKey('LastEdited', $dbFields);
 		$this->assertArrayNotHasKey('ClassName', $dbFields);
 		$this->assertArrayNotHasKey('ID', $dbFields);
+
+		// Assert that the correct field type is returned when passing a field
+		$this->assertEquals('Varchar', $obj->db('Name'));
+		$this->assertEquals('Text', $obj->db('Comment'));
```

Comment:
You'll also need to check that the 'Comment' key is the array returned by `$obj->db()` has the right type.